includes/class-newspack-popups-custom-placements.php

```diff
@@ -0,0 +1,256 @@
+	 * @param string        $fields Field values to return in response: 'all' or 'ids'.
+	 * @param array|boolean $categories Array of category IDs to filter by, empty array if fetching only uncategorized prompts, or false to ignore categories.
+	 *
+	 * @return array Array of prompt posts matching the custom placement.
+	 */
+	public static function get_prompts_for_custom_placement( $custom_placement_ids, $fields = 'all', $categories = false ) {
+		$args = [
+			'posts_per_page' => 100,
+			'post_status'    => 'publish',
+			'post_type'      => Newspack_Popups::NEWSPACK_POPUPS_CPT,
+			'fields'         => $fields,
+			'meta_key'       => 'placement',
+			'meta_compare'   => 'IN',
+			// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+			'meta_value'     => $custom_placement_ids,
+		];
+
+		if ( ! empty( $categories ) ) {
+			$args['category__in'] = $categories;
+		} elseif ( is_array( $categories ) ) {
+			// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+			$args['tax_query'] = [
+				[
+					'taxonomy' => 'category',
+					'operator' => 'NOT EXISTS',
+				],
+			];
+		}
+
+		$query = new \WP_Query( $args );
+
+		if ( $query->have_posts() ) {
+			return $query->posts;
+		}
+
+		return [];
+	}
+
+	/**
+	 * Validate a custom placement ID.
+	 *
+	 * @param string $custom_placement_id The slug of the custom placement to check.
+	 * @return string|boolean The ID if it's valid, false if not.
+	 */
+	public static function validate_custom_placement_id( $custom_placement_id ) {
+		$custom_placement_id = sanitize_text_field( $custom_placement_id );
+		return in_array( $custom_placement_id, self::get_custom_placement_values() ) ? $custom_placement_id : false;
+	}
+
+	/**
+	 * Get default placements that exist for all sites.
+	 */
+	public static function get_default_custom_placements() {
+		return [
+			'custom1' => __( 'Custom Placement 1' ),
+			'custom2' => __( 'Custom Placement 2' ),
+			'custom3' => __( 'Custom Placement 3' ),
+		];
+	}
+
+	/**
+	 * Get all configured placements.
+	 *
+	 * @return array Array of placements.
+	 */
+	public static function get_custom_placements() {
+		$placements = get_option( self::PLACEMENTS_OPTION_NAME, [] );
+		return array_merge( self::get_default_custom_placements(), $placements );
+	}
+
+	/**
+	 * Get a simple array of placement values.
+	 *
+	 * @return array Array of placement values.
+	 */
+	public static function get_custom_placement_values() {
+		return array_keys( self::get_custom_placements() );
+	}
+
+	/**
+	 * Determine whether the given prompt should be displayed via custom placement.
+	 *
+	 * @param object $prompt The prompt to assess.
+	 * @return boolean Whether or not the prompt has a custom placement.
+	 */
+	public static function is_custom_placement( $prompt ) {
+		return (
+			'manual' === $prompt['options']['frequency'] ||
+			in_array( $prompt['options']['frequency'], self::get_custom_placement_values() )
```

Comment:
Shouldn't this be looking at `placement`, instead of `frequency` option?

Reply:
True! 🤦 Fixed in f0b4679